client/tests/integration/tx_chainid.rs

```diff
@@ -0,0 +1,64 @@
+use std::str::FromStr;
+
+use iroha_crypto::KeyPair;
+use iroha_data_model::prelude::*;
+use test_network::*;
+
+use crate::integration::asset::asset_id_new;
+
+#[test]
+fn send_tx_with_different_chain_id() {
+    let (_rt, _peer, test_client) = <PeerBuilder>::new().with_port(11_240).start_with_runtime();
+    wait_for_genesis_committed(&[test_client.clone()], 0);
+    // Given
+    let sender_account_id = AccountId::from_str("sender@wonderland").expect("Valid");
+    let sender_keypair = KeyPair::generate().expect("Failed to generate sender KeyPair.");
+    let receiver_account_id = AccountId::from_str("receiver@wonderland").expect("Valid");
+    let asset_definition_id = AssetDefinitionId::from_str("test_asset#wonderland").expect("Valid");
+    let to_transfer = 1;
+    let create_sender_accout = Register::account(Account::new(
+        sender_account_id.clone(),
+        [sender_keypair.public_key().clone()],
+    ));
+    let create_receiver_account = Register::account(Account::new(receiver_account_id.clone(), []));
+    let create_asset =
+        Register::asset_definition(AssetDefinition::quantity(asset_definition_id.clone()));
+    let register_asset: InstructionBox = Register::asset(Asset::new(
+        AssetId::new(asset_definition_id.clone(), sender_account_id.clone()),
+        AssetValue::Quantity(10),
+    ))
+    .into();
```

Comment:
Quite misleading variable names. I suggest this:

```suggestion
    let register_asset_definition =
        Register::asset_definition(AssetDefinition::quantity(asset_definition_id.clone()));
    let register_asset: InstructionBox = Register::asset(Asset::new(
        AssetId::new(asset_definition_id.clone(), sender_account_id.clone()),
        AssetValue::Quantity(10),
    ))
    .into();
```